### PR TITLE
Refactor(plugins): Move jinja filter code for arista.avd.range_expand to PyAVD

### LINKED ISSUE #4137: Refactor(plugins): Move jinja filter code for arista.avd.range_expand to PyAVD
### Enhancement summary

Move jinja filter code for arista.avd.range_expand to PyAVD

### Which component of AVD is impacted

plugins

### Use case example

-

### Describe the solution you would like

The code and unit tests for range_expand filter should be moved to pyavd/j2filters and imported inside plugins/filter using pyavd wrapper.

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by laxmikantchintakindi**
## Change Summary

Move jinja filter code for arista.avd.range_expand to PyAVD

## Related Issue(s)

Fixes #4137

## Component(s) name

`arista.avd.range_expand`

## Proposed changes
Moving the code and unit tests for arista.avd.range_expand filter to PyAVD

### User Checklist

- N/A

### Repository Checklist

- [ ] My code has been rebased from devel before I start
- [ ] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [ ] My change requires a change to the documentation and documentation have been updated accordingly.
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/plugins/filter/range_expand.py [hunk 1]
```diff
@@ -6,9 +6,23 @@
 #
 from __future__ import absolute_import, division, print_function
 
+from ansible.errors import AnsibleFilterError
```
Comment (MaheshGSLAB) [inline ansible_collections/arista/avd/plugins/filter/range_expand.py]: same import at line 27

Comment (laxmikantchintakindi) [reply ansible_collections/arista/avd/plugins/filter/range_expand.py]: Fixed.

File: python-avd/tests/pyavd/j2filters/test_range_expand.py [hunk 2]
```diff
@@ -1,39 +1,37 @@
 # Copyright (c) 2023-2024 Arista Networks, Inc.
 # Use of this source code is governed by the Apache License 2.0
 # that can be found in the LICENSE file.
-from __future__ import absolute_import, division, print_function
 
-__metaclass__ = type
+from __future__ import absolute_import, division, print_function
```
Comment (ClausHolbechArista) [inline python-avd/tests/pyavd/j2filters/test_range_expand.py]:
```suggestion
from __future__ import annotations
```

File: ansible_collections/arista/avd/plugins/filter/range_expand.py [hunk 3]
```diff
@@ -6,11 +6,24 @@
...
+
 __metaclass__ = type
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/plugins/filter/range_expand.py]:
```suggestion
__metaclass__ = type

from ansible.errors import AnsibleFilterError
```

**Review by gmuloc — APPROVED** [review]

**Review by MaheshGSLAB — APPROVED** [review]

**Review by ClausHolbechArista — APPROVED** [review]

**Pull request merged**